src/InputWithLabel.jsx

```diff
@@ -0,0 +1,33 @@
+import PropTypes from "prop-types";
+import { useEffect, useRef } from "react";
+
+const InputWithLabel = (props) => {
+    const { todoTitle, handleTitleChange, children } = props;
+    const inputRef = useRef();
+    useEffect(() => {
+        inputRef.current.focus();
+    });
+
+    return (
+        <>
+            <label htmlFor="todoTitle">{children}</label>
```

Comment:
To answer your FINAL question about children, this prop is whatever you put inside the component's opening and closing tag when you use it somewhere else.

In the case of this assignment, in AddTodoForm it is:
```
<InputWithLabel>
    Title:
</InputWithLabel>
```

So the value in `children` when it gets to this component is a string "Title:", which we then can stick into the label. In this example, we want this component to be re-usable in other places in our application. But in other locations, the title of the label should be different. What if we wanted to create an InputWithLabel for a second field in this form that should be labeled "Description: " so that the user can provide extra context about what their TODO item is (or some sub-tasks, or anything else useful)?

Normally when you utilize the `children` prop, it's so that you can nest other React components/HTML elements inside your component. Let's say I am creating a component `FancyButton` - I want all my fancy buttons to have the same background look, but the content inside the button will be different every time (i.e. a different icon, or maybe sometimes some text instead of an icon).

I would declare a FancyButton component that looks like:
```
function FancyButton({children}){
    // Any functions or hooks get used here
    return (
        <div className="btn-background">
            <button className="btn">
                {children}
            </button>
        </div>
    );
}
```

And then when I want to use my button elsewhere, I can pass whatever content I want inside of it (which might be another React component):
```
<FancyButton>
    <SomeIcon className="icon-stylings" />
</FancyButton>

... elsewhere in your code ...
 
<FancyButton>
    <SomeOtherIcon className="other-icon-stylings" />
</FancyButton>
```

For this assignment, we just pass a basic string. We can pass it any valid HTML as well, which includes bare strings. In this case, I would personally just pass the label as a prop called `label` or something instead, but this is another way of doing it.